src/ask-password/ask-password.c

```diff
@@ -232,9 +240,163 @@ static int parse_argv(int argc, char *argv[]) {
+                { "echo",         SD_JSON_VARIANT_STRING,        dispatch_echo_mode,            offsetof(MethodAskParameters, echo_mode),     0 },
+                VARLINK_DISPATCH_POLKIT_FIELD,
+                {}
+        };
+
+        Hashmap **polkit_registry = ASSERT_PTR(userdata);
+        MethodAskParameters p = {
+                .timeout_usec = DEFAULT_TIMEOUT_USEC,
+                .until_usec = UINT64_MAX,
+                .accept_cached = -1,
+                .push_cache = -1,
+                .echo_mode = _ECHO_MODE_INVALID,
+        };
+        int r;
+
+        assert(link);
+
+        r = sd_varlink_dispatch(link, parameters, dispatch_table, &p);
+        if (r != 0)
+                return r;
+
+        r = varlink_verify_polkit_async_full(
+                        link,
+                        /* bus= */ NULL,
+                        "io.systemd.ask-password.ask",
+                        /* details= */ NULL,
+                        /* good_user= */ FLAGS_SET(arg_flags, ASK_PASSWORD_USER) ? getuid() : UID_INVALID,
+                        /* flags= */ 0,
+                        polkit_registry);
+        if (r <= 0)
+                return r;
+
+        AskPasswordRequest req = {
+                .tty_fd = -EBADF,
+                .message = p.message ?: arg_message,
+                .icon = p.icon ?: arg_icon,
+                .id = p.id ?: arg_id,
+                .keyring = p.keyring ?: arg_key_name,
+                .credential = arg_credential_name,
+                .hup_fd = sd_varlink_get_input_fd(link),
+        };
+
+        if (p.timeout_usec != 0 && p.until_usec != 0)
+                req.until = MIN(usec_add(now(CLOCK_MONOTONIC), p.timeout_usec), p.until_usec);
+
+        /* If the timeout is set to zero, don't ask agents, just stick to cache */
+        SET_FLAG(arg_flags, ASK_PASSWORD_NO_AGENT, req.until == 0);
+
+        if (p.accept_cached >= 0)
+                SET_FLAG(arg_flags, ASK_PASSWORD_ACCEPT_CACHED, p.accept_cached);
+
+        if (p.push_cache >= 0)
+                SET_FLAG(arg_flags, ASK_PASSWORD_PUSH_CACHE, p.push_cache);
+
+        if (p.echo_mode >= 0) {
+                SET_FLAG(arg_flags, ASK_PASSWORD_ECHO, p.echo_mode == ECHO_ON);
+                SET_FLAG(arg_flags, ASK_PASSWORD_SILENT, p.echo_mode == ECHO_OFF);
+        }
+
+        _cleanup_strv_free_erase_ char **l = NULL;
+
+        r = ask_password_auto(&req, arg_flags, &l);
+        if (r == -EUNATCH)
+                return sd_varlink_error(link, "io.systemd.AskPassword.NoPasswordAvailable", NULL);
+        if (r == -ETIME)
+                return sd_varlink_error(link, "io.systemd.AskPassword.TimeoutReached", NULL);
+        if (r == -ECONNRESET) { /* POLLHUP on the varlink fd we passed in via .hup_fd */
+                sd_varlink_close(link);
+                return 1;
+        }if (r < 0)
+                return r;
+
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *vl = NULL;
+
```

Comment:
I (and I thought also you) do not like the coding style to insert an empty line like here. Though, not a matter, and @YHNdnzj likes such style.

Reply:
yeah, i am with you on this. will fix.